resources/js/Pages/Lessons/Show.vue

```diff
@@ -0,0 +1,120 @@
+<script setup>
+import {onMounted, ref} from "vue";
+import unityGameSetUp from '../../GameViewJava';
+import {Head, Link} from '@inertiajs/inertia-vue3';
+
+onMounted(() => {
+    unityGameSetUp();
+    window.addEventListener("stateFromUnity", (e) => receivedJsonSaveFromUnity(e));
+    window.addEventListener("requestData", (e) => requestLoadGameData(e));
+    window.addEventListener("StartChallenge", (e) => receivedChallengeFromUnity(e));
+})
+
+const props = defineProps({
+    lesson: Object,
+    course: Object,
+})
+
+let showChallengeWindow = ref(false);
+let resultWarning = ref(false);
+let challengeUserAnswer = ref("");
+let challengeAnswer = ref("");
+
+const receivedChallengeFromUnity = async (e) => {
+    challengeAnswer.value = e.detail.challenge;
+    console.log(challengeAnswer.value);
+    showChallengeWindow.value = true;
+}
+
+const checkAnswer = (e) => {
+    if (challengeUserAnswer.value.trim() === challengeAnswer.value) {
+        showChallengeWindow.value = false;
+        gameInstance.SendMessage('GameController', 'CorrectAnswer');
+    } else {
+        resultWarning.value = true;
+    }
+}
+
+const receivedJsonSaveFromUnity = async (e) => {
+    try {
+        await axios.post(route('courses.lessons.save', {
+            course: props.course,
+            lesson: props.lesson
+        }), {
+            game_state: e.detail.gamestate
+        });
+    } catch (e) {
+        window.alert("failed to add to database, error: ", +e);
```

Comment:
I think you need to remove the comma
```suggestion
        window.alert("failed to add to database, error: " + e);
```